src/main/java/org/jenkinsci/plugins/ssegateway/Endpoint.java

```diff
@@ -100,6 +100,24 @@ protected void init() throws ServletException {
         PluginServletFilter.addFilter(new SSEListenChannelFilter());
     }
 
+    /**
+     * Returns the root directory for sse-events based on the Jenkins logs directory
+     * (historically always found under <code>$JENKINS_HOME/logs</code>.
+     * Configurable since Jenkins 2.114.
+     *
+     * @see hudson.triggers.SafeTimerTask#LOGS_ROOT_PATH_PROPERTY
+     * @return the root directory for logs.
+     */
+    private File getHistoryDirectory(Jenkins jenkins) {
+        File jenkinsLogsDir = new File(jenkins.getRootDir(), "/logs");
+
+        final String overriddenLogsRoot = System.getProperty("hudson.triggers.SafeTimerTask.logsTargetDir");
+        if (overriddenLogsRoot != null) {
+            jenkinsLogsDir = new File(overriddenLogsRoot);
+        }
+        return new File(jenkinsLogsDir, "sse-events");
+    }
```

Comment:
Or maybe more simply ....

```java
private File getHistoryDirectory(Jenkins jenkins) {
    final String overriddenLogsRoot = System.getProperty("hudson.triggers.SafeTimerTask.logsTargetDir");
    if (overriddenLogsRoot != null) {
        return new File(overriddenLogsRoot, "sse-events");
    } else {
        return new File(jenkins.getRootDir(), "/logs/sse-events");
    }
}
```

Reply:
heh, :man_facepalming: yeah. I guess it was late :P. Simpler, clearer, and much closer and consistent with what I did on the support-core plugin :) https://github.com/jenkinsci/support-core-plugin/pull/140

Fixing. Thanks!